packages/cli-lib/lib/interpolation.js

```diff
@@ -0,0 +1,139 @@
+const { helpers } = require('./text');
+const delimiters = {
+  interpolation: {
+    start: '{{',
+    end: '}}',
+  },
+  helpers: {
+    start: '#',
+    end: '/',
+  },
+};
+
+const isHelperIdentifier = identifier => {
+  return (
+    identifier.startsWith(delimiters.helpers.start) ||
+    identifier.startsWith(delimiters.helpers.end)
+  );
+};
+
+const generateReplaceFn = (matchedText, startIndex, replacementString) => {
+  return theString => {
+    const newString = `${theString.slice(0, startIndex)}${replacementString ||
+      ''}${theString.slice(startIndex + matchedText.length)}`;
+    return newString;
+  };
+};
+
+/**
+ * Interpolate a string with data
```

Comment:
I appreciate the detailed notes for these functions. I was going to suggest perhaps combining `interpolation` and `compileHelper`, since they're so similar, but honestly, I think that keeping them separate helps with readability.